src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -14,266 +12,46 @@
-                string[] functionFiles = fileSystem.Directory.EnumerateFiles(scriptDirectory)
-                    .Where(p => fileSystem.Path.GetFileName(p).ToLowerInvariant() != ScriptConstants.FunctionMetadataFileName)
-                    .ToArray();
-
-                if (functionFiles.Length == 0)
-                {
-                    throw new FunctionConfigurationException("No function script files present.");
-                }
-
-                if (functionFiles.Length == 1)
-                {
-                    // if there is only a single file, that file is primary
-                    functionPrimary = functionFiles[0];
-                }
-                else
-                {
-                    // if there is a "run" file, that file is primary,
-                    // for Node, any index.js file is primary
-                    functionPrimary = functionFiles.FirstOrDefault(p =>
-                        fileSystem.Path.GetFileNameWithoutExtension(p).ToLowerInvariant() == "run" ||
-                        fileSystem.Path.GetFileName(p).ToLowerInvariant() == "index.js");
-                }
-            }
-
-            if (string.IsNullOrEmpty(functionPrimary))
-            {
-                throw new FunctionConfigurationException("Unable to determine the primary function script. Try renaming your entry point script to 'run' (or 'index' in the case of Node), " +
-                    "or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.");
-            }
-
-            return Path.GetFullPath(functionPrimary);
-        }
-
-        internal static string ParseLanguage(string scriptFilePath, IEnumerable<WorkerConfig> workerConfigs)
-        {
-            // determine the script type based on the primary script file extension
-            string extension = Path.GetExtension(scriptFilePath).ToLowerInvariant().TrimStart('.');
-            switch (extension)
-            {
-                case "csx":
-                case "cs":
-                    return DotNetScriptTypes.CSharp;
-                case "dll":
-                    return DotNetScriptTypes.DotNetAssembly;
-            }
-            var workerConfig = workerConfigs.FirstOrDefault(config => config.Extensions.Contains("." + extension));
-            if (workerConfig != null)
-            {
-                return workerConfig.Language;
+                _logger.LogInformation($"A function whitelist has been specified, excluding all but the following functions: [{string.Join(", ", functionsWhiteList)}]");
+                metadata = metadata.Where(function => functionsWhiteList.Any(functionName => functionName.Equals(function.Name, StringComparison.CurrentCultureIgnoreCase))).ToImmutableArray();
+                Errors = _functionMetadataProvider.FunctionErrors.Where(kvp => functionsWhiteList.Contains(kvp.Key)).ToImmutableDictionary<string, ImmutableArray<string>>();
```

Comment:
The code above performs a case insensitive comparison using current culture, this doesn’t. Should these be consistent?

Reply:
You are right we should do case-insensitive comparison here too